src/spac/transformations.py

```diff
@@ -100,6 +98,12 @@ def get_cluster_info(
     else:
         check_feature(adata, features=features)
 
+    # Convert data matrix or specified layer to DataFrame using to_df()
+    if layer:
+        data_df = adata.to_df(layer=layer)

```

Comment:
@fangliu117 , please check the the passed argument "layer" is valid before using it.